### PR TITLE
Fix linting errors

### PR DESCRIPTION
**Pull request by nabramow**
Looks like a branch got merged with some linting errors. Just quickly fixing these because I want to be able to check the "no linting errors" box in my PR ;).

**Web frontend checklist**
- [ x ] Formatted my code with `yarn format`
- [ x ] There are no warnings from `yarn lint --fix`
- [ x ] There are no console warnings when running the app
- [ N/A ] Added any new components to storybook
- [ N/A ] Added tests where relevant
- [ x ] All tests pass
- [ x ] Clicked around my changes running locally and it works
- [ x ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by aapeliv** [comment]
Is @bakeiro working on these?

**Comment by bakeiro** [comment]
oh, I see... are were these errors  or warnings? (out of curiosity)

**Comment by bakeiro** [comment]
ah I though I executed that in my PR... seems that I mess up with `git stash` and the merge conflicts...

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.tsx:63]: why const rather normal function? (just to understand)

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.tsx:75]: is the behavior here the same?

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.tsx:88]: why not undefined?

File: app/web/features/search/SearchPage.tsx
Comment (bakeiro) [inline app/web/features/search/SearchPage.tsx]: same here, I remember I did this because when I was debugging the backend didn't receive the same values... is this checked?

File: app/web/features/search/SearchResultsList.test.tsx
Comment (bakeiro) [inline app/web/features/search/SearchResultsList.test.tsx:23]: 👍

File: app/web/features/search/SearchPage.tsx
Comment (bakeiro) [reply app/web/features/search/SearchPage.tsx:110]: ok I checked and backend receive the same values... all in order!

Comment (bakeiro) [reply app/web/features/search/SearchPage.tsx]: exactly, this creates a bug, there is a reason why I did this way, with your change if you use the lastActive filter (for example want to search lastActive "last day" and then want to unset this filter creates a bug)

image

Comment (bakeiro) [reply app/web/features/search/SearchPage.tsx]: that's why I created another issue (https://github.com/Couchers-org/couchers/issues/4565) and wanted to fix it there plus implement pagination

Comment (bakeiro) [reply app/web/features/search/SearchPage.tsx]: so please restore :)

File: app/web/features/search/FilterDialog.stories.tsx
Comment (bakeiro) [inline app/web/features/search/FilterDialog.stories.tsx:14]: 👍

File: app/web/features/search/FilterDialog.tsx
Comment (bakeiro) [inline app/web/features/search/FilterDialog.tsx:90]: thanks, I always use `any` when I try to set the type multiple times and fail (and after I forget to change it 😅)

File: app/web/features/search/SearchResultsList.tsx
Comment (bakeiro) [inline app/web/features/search/SearchResultsList.tsx]: 🔝

File: app/web/features/search/SearchBox.tsx
Comment (nabramow) [reply app/web/features/search/SearchBox.tsx:63]: It's not a huge deal either way, const seems a bit more modern to me.

Technically, using const creates a function expression which has a few advantages:

1. block-scoped behavior - meaning the function is only available within the block it's defined in, which gives better control over where the function can be accessed.
2. prevents re-declaration - using const guarantees the function can't be accidentally re-declared or reassigned within the same block, which adds a layer of safety.

Both are valid though!

File: app/web/features/search/SearchPage.tsx
Comment (nabramow) [reply app/web/features/search/SearchPage.tsx]: Gotcha sorry about that will leave it to you then, I put it back with a ts-ignore and to-do comment.

File: app/web/features/search/FilterDialog.tsx
Comment (nabramow) [reply app/web/features/search/FilterDialog.tsx:90]: Haha I gotcha, TS can be a pain sometimes.

I think some of the linting PRs I just opened will hopefully make it easier not to forget. I was also losing track of mine and only saw after Gitlab ran.

File: app/web/features/search/SearchBox.tsx
Comment (nabramow) [reply app/web/features/search/SearchBox.tsx:75]: You're right, I changed this back to undefined and updated the type to allow it. Sorry about that!

I got a little trigger happy last night...

Rather than making just the location undefined though, I'm returning the locationResult as undefined. That's what you had in the type originally anyway.

I did a lot of testing on the map and all seems to be working okay.

File: app/web/features/search/SearchBox.tsx
Comment (nabramow) [reply app/web/features/search/SearchBox.tsx:88]: Same as comment above, change whole object to undefined initially now.

File: app/web/features/search/FilterDialog.tsx
Comment (bakeiro) [inline app/web/features/search/FilterDialog.tsx]: 😮 no ts complains?

File: app/web/features/search/MapWrapper.tsx
Comment (bakeiro) [inline app/web/features/search/MapWrapper.tsx]: good find

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.tsx]: I don't think this is what I wanted, I think I wanted to delete all the values of location but not bbox

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.tsx]: same here

File: app/web/features/search/FilterDialog.tsx
Comment (nabramow) [reply app/web/features/search/FilterDialog.tsx]: It's because I set the type in the useState variable it references so now it matches :-)

File: app/web/features/search/SearchBox.tsx
Comment (nabramow) [reply app/web/features/search/SearchBox.tsx]: What behavior would that look like in the app? 

Since bbox represents a location, when would you want a bbox but not a location?

Comment (nabramow) [reply app/web/features/search/SearchBox.tsx]: Also on that note, do you know what triggers this particular function? I tried every form field in the modal, but nothing made this function run. I'm wondering if maybe it's old?

**Comment by nabramow** [comment]
Hey @bakeiro when you get a chance could you see if the map works as you expect with the undefined locationResult vs. just location undefined? It looks okay to me but maybe you see something I don't.

The type for `GeocodeResult` doesn't allow the `location` key to be undefined, so whoever wrote that originally didn't intend for it to be undefined. Especially since it's value is in instance of the `LngLat` class, which only accepts non-null numbers, looks like it's not supposed to be undefined.

We could change the type but I guess I just want to understand what you're trying to achieve with the `location: undefined` first.

File: app/web/features/search/FilterDialog.stories.tsx
Comment (bakeiro) [inline app/web/features/search/FilterDialog.stories.tsx]: why not undefined? I want to be able to have undefined, since it's the default value of the dropdown, plus it's what is sent to the server

File: app/web/features/search/MapWrapper.tsx
Comment (bakeiro) [inline app/web/features/search/MapWrapper.tsx]: 👍

File: app/web/features/search/MapWrapper.tsx
Comment (bakeiro) [inline app/web/features/search/MapWrapper.tsx]: 🔝

File: app/web/features/search/MapWrapper.tsx
Comment (bakeiro) [inline app/web/features/search/MapWrapper.tsx]: 👍

File: app/web/features/search/FilterDialog.tsx
Comment (bakeiro) [inline app/web/features/search/FilterDialog.tsx:88]: I get confused here, when there is no query I want to send undefined to the server... should be allowed to have undefined type then?

File: app/web/features/search/SearchBox.test.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.test.tsx:29]: thanks for this (and sorry for my anys!!)

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.tsx:45]: awesome!

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [reply app/web/features/search/SearchBox.tsx:63]: 📝📝📝 good info... will keep it!!

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [reply app/web/features/search/SearchBox.tsx]: it's the radio button on the form, available on desktop, basically let you choose between option a "location" and b "keyword" basically when user clicks keyword since the input of the location isn't defined I want to delete just the location, but keep the other values

Comment (bakeiro) [reply app/web/features/search/SearchBox.tsx]: as I suspected, this gets broken with that change 😅 (🤷‍♂️) 
image

Comment (bakeiro) [reply app/web/features/search/SearchBox.tsx]: don't worry, I make it work plus keepting the new types of this PR, I take care of this

File: app/web/features/search/SearchPage.test.tsx
Comment (bakeiro) [inline app/web/features/search/SearchPage.test.tsx]: 🙌 🙌

File: app/web/features/search/SearchBox.tsx
Comment (bakeiro) [inline app/web/features/search/SearchBox.tsx]: same here, can't set a location to undefined (sorry for the hassle, you know, need to maintain the page behavior 😅 )

Comment (bakeiro) [reply app/web/features/search/SearchBox.tsx]: don't worry, I will update it to make like before but with the new types

File: app/web/features/search/SearchPage.tsx
Comment (bakeiro) [inline app/web/features/search/SearchPage.tsx]: this is cool, what happens is that in the `mapWrapper` component it has some hardcoded coordinates which are the initial value (independently of the original state value)

that value is used by default, unless there is another value (when a user searches from the dashboard for example) then after the first load the bbox values are loaded, so it has no effect on the values that are loaded by default. (I feel bad typing this comments but it's how the map is wired 😅)

Comment (bakeiro) [reply app/web/features/search/SearchPage.tsx]: image

File: app/web/features/search/FilterDialog.stories.tsx
Comment (nabramow) [reply app/web/features/search/FilterDialog.stories.tsx]: I guess my confusion is whoever originally wrote the GeocodeResult type did not allow ‘location’ to be undefined. That’s why you’re having issues with TS, as it’s not allowed to be undefined according to the type.

The question is, did the person who wrote the original type do that for a reason? I assumed so, but maybe not.

In any case I’d say if you decide it should definitely be allowed to be undefined, then update the location key of the Geocode result type to allow undefined. That will make your life easier everywhere else.

File: app/web/features/search/SearchBox.tsx
Comment (nabramow) [reply app/web/features/search/SearchBox.tsx]: Sure, feel free to take over this PR. Probably easier than copy pasting and merge conflicts!

File: app/web/features/search/FilterDialog.tsx
Comment (nabramow) [reply app/web/features/search/FilterDialog.tsx:88]: Okay looking at this more I think you are right. My original thinking was using "" could help ensure a consistent type for the queryName field. Generally text fields represent being empty by an empty string.

But, when I looked in app/web/service/search.ts, which is processing this search, on line 35 it's checking if `query !== undefined` to decide whether to include it.

Maybe a quirk of GRPC or React Query, I'm just used to it being set up a bit differently.

Feel free to add undefined sorry about that! Still learning this codebase.

File: app/web/features/search/SearchPage.tsx
Comment (nabramow) [reply app/web/features/search/SearchPage.tsx]: No worries maps are complicated.

File: app/web/features/search/FilterDialog.tsx
Comment (bakeiro) [reply app/web/features/search/FilterDialog.tsx:88]: True, I could use an empty string and then if empty string then send an undefined to the server... that's also right :) [I'm my case I'm more used to undefined = something not set, and "" as empty string (but actual value)

File: app/web/features/search/FilterDialog.stories.tsx
Comment (bakeiro) [reply app/web/features/search/FilterDialog.stories.tsx]: first sorry for late reply
second, I think we are confusing `numberOfGuestFilter` with Geocoderesult, I'm with you that never should be undefined, I was talking about `numberOfGuestFilter` :)

File: app/web/features/search/FilterDialog.tsx
Comment (bakeiro) [reply app/web/features/search/FilterDialog.tsx:88]: actually you are right... it's more stable having 1 type, that 2, makes code more "robust" in my opinion... keeping your change!

**Review by bakeiro — COMMENTED** [review]
.

File: app/web/features/search/SearchResultsList.tsx
Comment (nabramow) [inline app/web/features/search/SearchResultsList.tsx]: I believe you can turn this into one variable like:

```javascript
const wasResultFound = resultsList?.find((value) => value.user?.userId === selectedResult?.userId) !== undefined;
```

It's a bit more performant as `find` will stop as soon as the first value is true:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: app/web/features/search/SearchResultsList.tsx
Comment (nabramow) [inline app/web/features/search/SearchResultsList.tsx:132]: Do we have any filters right now that take `rank` or `snippet` into account? I was also wondering this and if it could sort someone wrong due to that.

File: app/web/service/search.ts
Comment (nabramow) [inline app/web/service/search.ts]: Looks like you can just do:

```javascript
if (query) {}
```

https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Screenshot 2024-09-17 at 9 29 10 AM

Screenshot 2024-09-17 at 9 29 15 AM

File: app/web/features/search/SearchResultsList.tsx
Comment (nabramow) [inline app/web/features/search/SearchResultsList.tsx]: I believe you can turn this into one variable like:

```javascript
const wasResultFound = resultsList?.find((value) => value.user?.userId === selectedResult?.userId) !== undefined;
```

It's a bit more performant as find will stop as soon as the first value is true:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

File: app/web/service/search.ts
Comment (bakeiro) [reply app/web/service/search.ts]: cool!!!

Comment (bakeiro) [reply app/web/service/search.ts]: updated ✅

Comment (bakeiro) [reply app/web/service/search.ts]: I've spend to much time learning these things on PHP that I forgot to learn it about JS

File: app/web/features/search/SearchResultsList.tsx
Comment (bakeiro) [reply app/web/features/search/SearchResultsList.tsx:132]: I think should be fine since this only happens when the result isn't on the list, so we are displaying only one result (basically the first time map is loaded all the users are shown, but due pagination we show only a subset in the list (only on desktop)

this shows the user when the user clicks on it on the map, so list would be only 1 result, so I don't think the hardcoded rank will be a problem :)

File: app/web/features/search/SearchResultsList.tsx
Comment (bakeiro) [reply app/web/features/search/SearchResultsList.tsx]: I guess that instead of `!== undefined` should be `false`?? (if I'm understanding properly)

Comment (nabramow) [reply app/web/features/search/SearchResultsList.tsx]: .find will return undefined if nothing passes the given test. Since you want it to return a boolean I added the !== undefined. So if it found something then it’s not equal to undefined it will return true so wasResultFound = true.

If it does return undefined then undefined !== undefined will return false so wasResultFound = false.

Does that make sense? Not sure I explained well.

Comment (bakeiro) [reply app/web/features/search/SearchResultsList.tsx]: sure, absolutely clear, will update it now!

Comment (bakeiro) [reply app/web/features/search/SearchResultsList.tsx]: done ✅

**Review by bakeiro — DISMISSED** [review]
should be fine now :)

**Comment by nabramow** [comment]
@bakeiro Looks like the prettier format is failing. If you fix that and gitlab is passing happy for you merge!

**Review by bakeiro — APPROVED** [review]

**Comment by bakeiro** [comment]
pending testing, do not merge yet

**Pull request merged**
